modular_skyrat/modules/xenos_skyrat_redo/code/xeno_types/ravager.dm

```diff
@@ -0,0 +1,140 @@
+	/// Holds the endure ability to be granted to the ravager later
+	var/datum/action/cooldown/alien/skyrat/literally_too_angry_to_die/you_cant_hurt_me_jack
+	melee_damage_lower = 30
+	melee_damage_upper = 35
+
+/mob/living/carbon/alien/humanoid/skyrat/ravager/Initialize(mapload)
+	. = ..()
+	triple_charge = new /datum/action/cooldown/mob_cooldown/charge/triple_charge/ravager()
+	triple_charge.Grant(src)
+
+	tailsweep_slice = new /datum/action/cooldown/spell/aoe/repulse/xeno/slicing()
+	tailsweep_slice.Grant(src)
+
+	you_cant_hurt_me_jack = new /datum/action/cooldown/alien/skyrat/literally_too_angry_to_die
+	you_cant_hurt_me_jack.Grant(src)
+
+	REMOVE_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/mob/living/carbon/alien/humanoid/skyrat/ravager/Destroy()
+	QDEL_NULL(triple_charge)
+	QDEL_NULL(tailsweep_slice)
+	QDEL_NULL(you_cant_hurt_me_jack)
+	return ..()
+
+/mob/living/carbon/alien/humanoid/skyrat/ravager/create_internal_organs()
+	internal_organs += new /obj/item/organ/internal/alien/plasmavessel
+	..()
+
+/datum/action/cooldown/mob_cooldown/charge/triple_charge/ravager
+	name = "Triple Charge Attack"
+	desc = "Allows you to charge thrice at a location, trampling any in your path."
+	cooldown_time = 30 SECONDS
+	charge_delay = 0.3 SECONDS
+	charge_distance = 7
+	charge_past = 3
+	destroy_objects = FALSE
+	charge_damage = 50
+	icon_icon = 'modular_skyrat/modules/xenos_skyrat_redo/icons/xeno_actions.dmi'
+	button_icon_state = "ravager_charge"
+	unset_after_click = TRUE
+
+/datum/action/cooldown/mob_cooldown/charge/triple_charge/ravager/do_charge_indicator(atom/charger, atom/charge_target)
+	playsound(charger, 'modular_skyrat/modules/xenos_skyrat_redo/sound/alien_roar2.ogg', 100, TRUE, 8, 0.9)
+
+/datum/action/cooldown/mob_cooldown/charge/triple_charge/ravager/Activate(atom/target_atom)
+	. = ..()
+	return TRUE
+
+/datum/action/cooldown/spell/aoe/repulse/xeno/slicing
+	name = "Slicing Tail Sweep"
+	desc = "Throw back attackers with a swipe of your tail, slicing them with it's sharpened tip."
+
+	cooldown_time = 60 SECONDS
+
+	aoe_radius = 2
+
+	icon_icon = 'modular_skyrat/modules/xenos_skyrat_redo/icons/xeno_actions.dmi'
+	button_icon_state = "slice_tail"
+
+	sparkle_path = /obj/effect/temp_visual/dir_setting/tailsweep/ravager
+
+	sound = 'modular_skyrat/modules/xenos_skyrat_redo/sound/alien_tail_swipe.ogg' //The defender's tail sound isn't changed because its big and heavy, this isn't
+
+/datum/action/cooldown/spell/aoe/repulse/xeno/slicing/cast_on_thing_in_aoe(atom/movable/victim, atom/caster)
+	if(isalien(victim))
+		return
+	var/turf/throwtarget = get_edge_target_turf(caster, get_dir(caster, get_step_away(victim, caster)))
+	var/dist_from_caster = get_dist(victim, caster)
+
+	if(dist_from_caster <= 0)
+		if(isliving(victim))
+			var/mob/living/victim_living = victim
+			victim_living.Knockdown(10 SECONDS)
+			victim_living.apply_damage(40, BRUTE,BODY_ZONE_CHEST, wound_bonus = 20, sharpness = SHARP_EDGED)
```

Comment:
There's so much copy-paste of this code, I think it should be made into a proc so that it's not copy-pasted in every single one of those files.